datalad/distributed/create_sibling_ghlike.py

```diff
@@ -0,0 +1,638 @@
+                recursive=False,
+                # TODO fetch=True, maybe only if one existed already
+                publish_depends=publish_depends,
+                result_renderer='disabled')
+
+    def repo_create_request(self, reponame, organization, private,
+                            dry_run=False):
+        """Perform a request to create a repo on the target platform
+
+        Also implements reporting of "fake" results in dry-run mode.
+
+        Returns
+        -------
+        dict
+          Result record, but see repo_create_response() for details.
+        """
+        endpoint = urljoin(
+            self.api_url,
+            self.create_org_repo_endpoint.format(
+                organization=organization)
+            if organization else
+            self.create_user_repo_endpoint)
+        data = {
+            'name': reponame,
+            'description': 'some default',
+            'private': private,
+            'auto_init': False,
+        }
+        headers = self.request_headers
+
+        if dry_run:
+            return dict(
+                status='ok',
+                request_url=endpoint,
+                request_data=data,
+                request_headers=headers,
+            )
+        r = requests.post(
+            endpoint,
+            json=data,
+            headers=headers,
+        )
+        return self.repo_create_response(r)
+
+    def repo_create_response(self, r):
+        """Handling of repo creation request responses
+
+        Normalizes error handling and reporting.
+
+        Returns
+        -------
+        dict
+          Result record
+
+        Raises
+        ------
+        Exception
+          Raises for any unhandles HTTP error response code.
+        """
+        try:
+            response = r.json()
+        except Exception as e:
+            lgr.debug('Cannot get JSON payload of %s [%s]' , r, e)
+            response = {}
+        lgr.debug('%s responded with %s %s', self.fullname, r, response)
+        if r.status_code == self.response_code_repo_created:
+            return dict(
+                status='ok',
+                preexisted=False,
+                # return in full
+                host_response=response,
+                # perform some normalization
+                **self.normalize_repo_properties(response)
+            )
+        elif r.status_code == requests.codes.unprocessable and \
+                'already exist' in response.get('message', ''):
+            return dict(
+                status='impossible',
+                message='repository already exists',
+                preexisted=True,
+            )
+        elif r.status_code == self.response_code_unauthorized:
+            return dict(
+                status='error',
+                message=('unauthorized: %s', response.get('message')),
+            )
+        # make sure any error-like situation causes noise
+        r.raise_for_status()
```

Comment:
As pointed out in https://github.com/datalad/datalad/pull/5949#issuecomment-924713051, with that `raise_for_status` we kind of throw away the message from the platform (`str(r)` would have it), turning it into a generic HTTPError. Not sure how to it best, but we could simply enhance the message. Something along those lines:

```
try:
    r.raise_for_status()
except HTTPError as e:
    raise HTTPError(str(e) + '%s responded with %s %s'.format(self.fullname, r, response)) from e
```

`requests.Response` would also allow to overwrite the default `.reason` that is used to build the `HTTPError`, but that seems to much reliance on the reponses  body to be more meaningful.

Alternatively: Catch and return an error result? Not sure about implications down the line yet.

Last thing I can see ATM: `HTTPError` does have the `Response` linked already it seems. Then it could also be up to rendering the exception to display the actual response. Either way, just letting the generic HTTPError bubble up, swallows meaningful information.

Reply:
I decided to specifically support the error condition that you faced, rather than adding more sophisticated (re)raising of exceptions, that would nevertheless lead to debugging. Using a token without permissions to create a repo under an existing org now looks like:

```
create_sibling_gin(error): [unauthorized: given user is not owner of organization]
```